lib/queue.js

```diff
@@ -276,6 +276,37 @@ class Queue extends Emitter {
     });
   }
 
+  _getJobsByIds(ids) {
+    // If ids is a Set (i.e. returned from Queue._scanForJobs), convert it to an
+    // Array.
+    const idsArray = Array.from(ids);
+    if (!idsArray.length) {
+      return Promise.resolve([]);
+    }
+
+    // We need to re-ensure the queue is commandable, as we might be shutting
+    // down during this operation.
+    return this._commandable().then((client) => {
+      const got = helpers.deferred();
+      const commandArgs = [this.toKey('jobs')].concat(idsArray, got.defer());
```

Comment:
Although on second thought, maybe in that case we would want to have disabled `storeJobs` on that queue instance anyway to avoid memory issues, and so this wouldn't make any difference. I guess the "watch jobs that have failed" use case wouldn't be so sensitive to it assuming we don't end up with many failed jobs, but for the "watch jobs that are waiting" use case we would almost certainly want to have `storeJobs` disabled.